lib/sync.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const utils = require('./utils');
+const noop = value => () => value;
+
+const kPath = Symbol('path');
+const kRealPathExists = Symbol('realpath-exists');
+const kResult = Symbol('result');
+
+const defaults = {
+  depth: Infinity,
+  objects: false,
+  realpath: false,
+  recursive: false,
+
+  onDirectory: null,
+  onEach: null,
+  onEmpty: null,
+  onFile: null
+};
+
+const readdirSync = (basedir, options = {}) => {
+  if (Array.isArray(basedir)) {
+    return readdirsSync(basedir, options);
+  }
+
+  const opts = { ...defaults, ...options };
+
+  const isMaxDepth = file => file.depth >= opts.depth;
+  const isIgnoredFile = opts.isIgnoredFile || noop(false);
+  const isIgnoredDir = opts.isIgnoredDir || noop(false);
+  const isMatch = opts.isMatch ? utils.matcher(opts.isMatch, opts) : () => true;
+
+  const keepDirs = opts.dirs !== false && opts.nodir !== true;
+  const cwd = path.resolve(basedir);
+  const base = opts.base ? path.resolve(opts.base) : cwd;
+  const seen = new Set();
+  const results = [];
+
+  const symlinks = (opts.symlinks !== false && opts.follow !== false) || opts.realpath === true;
+  const follow = opts.follow === true || opts.realpath === true || opts.symlinks === true;
+  const recurse = opts.recursive === true || (typeof options.depth === 'number' && options.depth > 1);
+
+  const getReturnValue = file => {
+    if (file.keep === false) return;
+    if (file.ignore === true) return;
+
+    if (file.keep !== true) {
+      if (file.isSymbolicLink() && symlinks !== true) return;
+      if (file.isDirectory() && options.nodir === true) return;
+      if (options.dot === false && file.name[0] !== '.') return;
```

Comment:
Same question above...

```suggestion
      if (options.dot === false && file.name[0] === '.') return;
```